### PR TITLE
moved last vector kernels from hiopIterate into hiopVector

### PR DESCRIPTION
**Pull request by cnpetra**
Addresses and closes issue #151 

 - provides default and RAJA vectorized implementations
 - added tests in vector testing module
 - updated documentation

Tested on Summit.

### CODE REVIEW

**Comment by pelesh** [comment]
I am guessing `addLinearDampingTerm` is MPI distributed? If so, it would be good to document that with a comment in the code.

File: tests/LinAlg/vectorTests.hpp [hunk 1]
```diff
@@ -1085,6 +1085,98 @@ class VectorTests : public TestBase
...
+   */
+  bool vectorAddLinearDampingTerm(
+      hiop::hiopVector& x,
+      hiop::hiopVector& left,
+      hiop::hiopVector& right,
+      const int rank)
+  {
+    const local_ordinal_type N = getLocalSize(&x);
+    assert(N == getLocalSize(&left));
+    assert(N == getLocalSize(&right));
```
Comment (ashermancinelli) [inline tests/LinAlg/vectorTests.hpp:1101]: I think it would be reasonable to just require size >=4 for this test rather than checking in the test and possibly leaving some corner cases untested.

```suggestion
  {
    const local_ordinal_type N = getLocalSize(&x);
    assert(N == getLocalSize(&left));
    assert(N == getLocalSize(&right));
    assert(N >= 4 && "Test `vectorAddLinearDampingTerm` must have size >=4 to test all use cases for this kernel");
```

**Review by ashermancinelli — CHANGES_REQUESTED** [review]
lgtm! though if it's not too much work I suggest just adding an assertion and simplifying the test for `vectorAddLinearDampingTerm`

File: tests/LinAlg/vectorTests.hpp (hunk 1 above)
Comment (cnpetra) [reply tests/LinAlg/vectorTests.hpp:1101]: that way one cannot test vectors with unbalanced MPI distribution (as in potentially empty local data in an otherwise quite long vector)

Comment (ashermancinelli) [reply tests/LinAlg/vectorTests.hpp:1101]: Okay, how should I run the test in order to test each case? I would naively just run the test on a vector with size >=4, so I think documenting how the function must be called in order to fully test would be helpful
